diff --git a/src/ui/versionStatus.ts b/src/ui/versionStatus.ts
index 3e1629d..455f949 100644
--- a/src/ui/versionStatus.ts
+++ b/src/ui/versionStatus.ts
@@ -20,7 +20,8 @@ export class VersionStatus extends Disposable {
 	) {
 		super();
 
-		this._statusItem = this._register(vscode.languages.createLanguageStatusItem('typescript.version', jsTsLanguageModes));
+		// this._statusItem = this._register(vscode.languages.createLanguageStatusItem('typescript.version', jsTsLanguageModes));
+    this._statusItem = {} as any;
 
 		this._statusItem.name = vscode.l10n.t("TypeScript Version");
 		this._statusItem.detail = vscode.l10n.t("TypeScript Version");
