diff --git a/src/tsServer/versionProvider.electron.ts b/src/tsServer/versionProvider.electron.ts
index dca0fb2..190b8e2 100644
--- a/src/tsServer/versionProvider.electron.ts
+++ b/src/tsServer/versionProvider.electron.ts
@@ -10,6 +10,9 @@ import API from '../utils/api';
 import { TypeScriptServiceConfiguration } from '../utils/configuration';
 import { RelativeWorkspacePathResolver } from '../utils/relativePathResolver';
 import { ITypeScriptVersionProvider, TypeScriptVersion, TypeScriptVersionSource } from './versionProvider';
+import { createRequire } from 'module';
+import { fileURLToPath } from 'url';
+const req = createRequire(import.meta.url);
 
 
 export class DiskTypeScriptVersionProvider implements ITypeScriptVersionProvider {
@@ -78,14 +81,16 @@ export class DiskTypeScriptVersionProvider implements ITypeScriptVersionProvider
 
 	private getContributedVersion(source: TypeScriptVersionSource, extensionId: string, pathToTs: readonly string[]): TypeScriptVersion | undefined {
 		try {
-			const extension = vscode.extensions.getExtension(extensionId);
-			if (extension) {
-				const serverPath = path.join(extension.extensionPath, ...pathToTs, 'typescript', 'lib', 'tsserver.js');
+			// const extension = vscode.extensions.getExtension(extensionId);
+			// if (extension) {
+				const tsPath = req.resolve("typescript", { paths: [fileURLToPath(import.meta.url)] })
+				const serverPath = path.join(path.dirname(tsPath), 'tsserver.js');
+
 				const bundledVersion = new TypeScriptVersion(source, serverPath, DiskTypeScriptVersionProvider.getApiVersion(serverPath), '');
 				if (bundledVersion.isValid) {
 					return bundledVersion;
 				}
-			}
+			// }
 		} catch {
 			// noop
 		}
