diff --git a/src/tsServer/bufferSyncSupport.ts b/src/tsServer/bufferSyncSupport.ts
index 2bb4d39..c6f4d6f 100644
--- a/src/tsServer/bufferSyncSupport.ts
+++ b/src/tsServer/bufferSyncSupport.ts
@@ -786,10 +786,6 @@ export default class BufferSyncSupport extends Disposable {
 			return false;
 		}
 
-		if (!this.client.configuration.enableProjectDiagnostics && !this._tabResources.has(buffer.resource)) { // Only validate resources that are showing to the user
-			return false;
-		}
-
 		switch (buffer.languageId) {
 			case languageModeIds.javascript:
 			case languageModeIds.javascriptreact:
