diff --git a/src/tsServer/bufferSyncSupport.ts b/src/tsServer/bufferSyncSupport.ts
index 7b47be8..cea92d0 100644
--- a/src/tsServer/bufferSyncSupport.ts
+++ b/src/tsServer/bufferSyncSupport.ts
@@ -769,10 +769,6 @@ export default class BufferSyncSupport extends Disposable {
 			return false;
 		}
 
-		if (!this.client.configuration.enableProjectDiagnostics && !this._tabResources.has(buffer.resource)) { // Only validate resources that are showing to the user
-			return false;
-		}
-
 		switch (buffer.languageId) {
 			case languageModeIds.javascript:
 			case languageModeIds.javascriptreact:
