diff --git a/src/languageProvider.ts b/src/languageProvider.ts
index feb47f0..639a58d 100644
--- a/src/languageProvider.ts
+++ b/src/languageProvider.ts
@@ -92,6 +92,7 @@ export default class LanguageProvider extends Disposable {
 			import('./languageFeatures/tagClosing').then(provider => this._register(provider.register(selector, this.description, this.client))),
 			import('./languageFeatures/typeDefinitions').then(provider => this._register(provider.register(selector, this.client))),
 		]);
+		(vscode.languages as any).$triggerStaticFeaturesRegistered();
 	}
 
 	private configurationChanged(): void {
