diff --git a/src/languageFeatures/util/textRendering.ts b/src/languageFeatures/util/textRendering.ts
index af1a7e6..f5540a7 100644
--- a/src/languageFeatures/util/textRendering.ts
+++ b/src/languageFeatures/util/textRendering.ts
@@ -155,15 +155,12 @@ function convertLinkTags(
 			case 'link':
 				if (currentLink) {
 					if (currentLink.target) {
-						const file = filePathConverter.toResource(currentLink.target.file);
-						const args: OpenJsDocLinkCommand_Args = {
-							file: { ...file.toJSON(), $mid: undefined }, // Prevent VS Code from trying to transform the uri,
-							position: typeConverters.Position.fromLocation(currentLink.target.start)
-						};
-						const command = `command:${OpenJsDocLinkCommand.id}?${encodeURIComponent(JSON.stringify([args]))}`;
-
+						const link = filePathConverter.toResource(currentLink.target.file)
+							.with({
+								fragment: `L${currentLink.target.start.line},${currentLink.target.start.offset}`
+							});
 						const linkText = currentLink.text ? currentLink.text : escapeMarkdownSyntaxTokensForCode(currentLink.name ?? '');
-						out.push(`[${currentLink.linkcode ? '`' + linkText + '`' : linkText}](${command})`);
+						out.push(`[${currentLink.linkcode ? '`' + linkText + '`' : linkText}](${link.toString()})`);
 					} else {
 						const text = currentLink.text ?? currentLink.name;
 						if (text) {
