diff --git a/src/tsServer/server.ts b/src/tsServer/server.ts
index 0952950..b340701 100644
--- a/src/tsServer/server.ts
+++ b/src/tsServer/server.ts
@@ -3,7 +3,6 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { Cancellation } from '@vscode/sync-api-common/lib/common/messageCancellation';
 import * as vscode from 'vscode';
 import { TypeScriptServiceConfiguration } from '../configuration/configuration';
 import { TelemetryReporter } from '../logging/telemetry';
@@ -234,12 +233,12 @@ export class SingleTsServer extends Disposable implements ITypeScriptServer {
 
 				if (executeInfo.token) {
 
-					const cancelViaSAB = isWebAndHasSharedArrayBuffers()
-						? Cancellation.addData(request)
-						: undefined;
+					// const cancelViaSAB = isWebAndHasSharedArrayBuffers()
+					// 	? Cancellation.addData(request)
+					// 	: undefined;
 
 					executeInfo.token.onCancellationRequested(() => {
-						cancelViaSAB?.();
+						// cancelViaSAB?.();
 						this.tryCancelRequest(request, command);
 					});
 				}
