diff --git a/src/typeScriptServiceClientHost.ts b/src/typeScriptServiceClientHost.ts
index c44bfc3..0268e5f 100644
--- a/src/typeScriptServiceClientHost.ts
+++ b/src/typeScriptServiceClientHost.ts
@@ -30,9 +30,7 @@ import * as typeConverters from './typeConverters';
 import TypeScriptServiceClient from './typescriptServiceClient';
 import { ActiveJsTsEditorTracker } from './ui/activeJsTsEditorTracker';
 import { IntellisenseStatus } from './ui/intellisenseStatus';
-import * as LargeProjectStatus from './ui/largeProjectStatus';
 import TypingsStatus, { AtaProgressReporter } from './ui/typingsStatus';
-import { VersionStatus } from './ui/versionStatus';
 import { coalesce } from './utils/arrays';
 import { Disposable } from './utils/dispose';
 
@@ -97,11 +95,11 @@ export default class TypeScriptServiceClientHost extends Disposable {
 		this.client.onConfigDiagnosticsReceived(diag => this.configFileDiagnosticsReceived(diag), null, this._disposables);
 		this.client.onResendModelsRequested(() => this.populateService(), null, this._disposables);
 
-		this._register(new VersionStatus(this.client));
+		// this._register(new VersionStatus(this.client));
 		this._register(new IntellisenseStatus(this.client, services.commandManager, services.activeJsTsEditorTracker));
 		this._register(new AtaProgressReporter(this.client));
 		this.typingsStatus = this._register(new TypingsStatus(this.client));
-		this._register(LargeProjectStatus.create(this.client));
+		// this._register(LargeProjectStatus.create(this.client));
 
 		this.fileConfigurationManager = this._register(new FileConfigurationManager(this.client, onCaseInsensitiveFileSystem));
 
