diff --git a/src/typescriptServiceClient.ts b/src/typescriptServiceClient.ts
index b5fdb18..d196d68 100644
--- a/src/typescriptServiceClient.ts
+++ b/src/typescriptServiceClient.ts
@@ -877,7 +877,7 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 					expectsResult: true,
 					...config,
 				});
-				executions[0]!.finally(() => {
+				executions[0] = executions[0]!.finally(() => {
 					runningServerState.toCancelOnResourceChange.delete(inFlight);
 					source.dispose();
 				});
