diff --git a/src/typescriptServiceClient.ts b/src/typescriptServiceClient.ts
index a1f06a3..41d3419 100644
--- a/src/typescriptServiceClient.ts
+++ b/src/typescriptServiceClient.ts
@@ -881,7 +881,7 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 					expectsResult: true,
 					...config,
 				});
-				executions[0]!.finally(() => {
+				executions[0] = executions[0]!.finally(() => {
 					runningServerState.toCancelOnResourceChange.delete(inFlight);
 					source.dispose();
 				});
