diff --git a/src/typescriptServiceClient.ts b/src/typescriptServiceClient.ts
index d9f47c1..f45f4bf 100644
--- a/src/typescriptServiceClient.ts
+++ b/src/typescriptServiceClient.ts
@@ -868,7 +868,7 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 					expectsResult: true,
 					...config,
 				});
-				executions[0]!.finally(() => {
+				executions[0] = executions[0]!.finally(() => {
 					runningServerState.toCancelOnResourceChange.delete(inFlight);
 					source.dispose();
 				});
