diff --git a/src/typescriptServiceClient.ts b/src/typescriptServiceClient.ts
index 4201d6d..97ea40b 100644
--- a/src/typescriptServiceClient.ts
+++ b/src/typescriptServiceClient.ts
@@ -869,7 +869,7 @@ export default class TypeScriptServiceClient extends Disposable implements IType
 					expectsResult: true,
 					...config,
 				});
-				executions[0]!.finally(() => {
+				executions[0] = executions[0]!.finally(() => {
 					runningServerState.toCancelOnResourceChange.delete(inFlight);
 					source.dispose();
 				});
