diff --git a/src/commands/goToProjectConfiguration.ts b/src/commands/goToProjectConfiguration.ts
index 0222f32..a6e2172 100644
--- a/src/commands/goToProjectConfiguration.ts
+++ b/src/commands/goToProjectConfiguration.ts
@@ -8,6 +8,7 @@ import { ActiveJsTsEditorTracker } from '../ui/activeJsTsEditorTracker';
 import { Lazy } from '../utils/lazy';
 import { openProjectConfigForFile, ProjectType } from '../tsconfig';
 import { Command } from './commandManager';
+import * as vscode from "vscode";
 
 export class TypeScriptGoToProjectConfigCommand implements Command {
 	public readonly id = 'typescript.goToProjectConfig';
@@ -17,11 +18,11 @@ export class TypeScriptGoToProjectConfigCommand implements Command {
 		private readonly lazyClientHost: Lazy<TypeScriptServiceClientHost>,
 	) { }
 
-	public execute() {
-		const editor = this.activeJsTsEditorTracker.activeJsTsEditor;
-		if (editor) {
-			openProjectConfigForFile(ProjectType.TypeScript, this.lazyClientHost.value.serviceClient, editor.document.uri);
-		}
+	public execute(uri: string) {
+		// const editor = this.activeJsTsEditorTracker.activeJsTsEditor;
+		// if (editor) {
+			openProjectConfigForFile(ProjectType.TypeScript, this.lazyClientHost.value.serviceClient, vscode.Uri.parse(uri));
+		// }
 	}
 }
 
