diff --git a/src/languageFeatures/fileReferences.ts b/src/languageFeatures/fileReferences.ts
index 3a475ac..aa602c2 100644
--- a/src/languageFeatures/fileReferences.ts
+++ b/src/languageFeatures/fileReferences.ts
@@ -46,7 +46,7 @@ class FileReferencesCommand implements Command {
 			return;
 		}
 
-		await vscode.window.withProgress({
+		return await vscode.window.withProgress({
 			location: vscode.ProgressLocation.Window,
 			title: vscode.l10n.t("Finding file references")
 		}, async (_progress, token) => {
@@ -60,6 +60,7 @@ class FileReferencesCommand implements Command {
 
 			const locations: vscode.Location[] = response.body.refs.map(reference =>
 				typeConverters.Location.fromTextSpan(this.client.toResource(reference.file), reference));
+      return locations;
 
 			const config = vscode.workspace.getConfiguration('references');
 			const existingSetting = config.inspect<string>('preferredLocation');
