diff --git a/src/commands/tsserverRequests.ts b/src/commands/tsserverRequests.ts
index 50aeedf..96ab233 100644
--- a/src/commands/tsserverRequests.ts
+++ b/src/commands/tsserverRequests.ts
@@ -33,7 +33,8 @@ export class TSServerRequestCommand implements Command {
 			// Introspecting code at a position
 			'quickinfo',
 			'quickinfo-full',
-			'completionInfo'
+			'completionInfo',
+			'getMoveToRefactoringFileSuggestions'
 		];
 
 		if (allowList.includes(command) || command.startsWith('_')) {
diff --git a/src/languageFeatures/refactor.ts b/src/languageFeatures/refactor.ts
index 1317cda..3e28248 100644
--- a/src/languageFeatures/refactor.ts
+++ b/src/languageFeatures/refactor.ts
@@ -124,6 +124,7 @@ namespace MoveToFileRefactorCommand {
 		readonly action: Proto.RefactorActionInfo;
 		readonly range: vscode.Range;
 		readonly trigger: vscode.CodeActionTriggerKind;
+		readonly targetFile?: string;
 	}
 }
 
@@ -142,7 +143,7 @@ class MoveToFileRefactorCommand implements Command {
 			return;
 		}
 
-		const targetFile = await this.getTargetFile(args.document, file, args.range);
+		const targetFile = args.targetFile;
 		if (!targetFile || targetFile.toString() === file.toString()) {
 			return;
 		}
