diff --git a/src/languageFeatures/refactor.ts b/src/languageFeatures/refactor.ts
index 1317cda..3e28248 100644
--- a/src/languageFeatures/refactor.ts
+++ b/src/languageFeatures/refactor.ts
@@ -124,6 +124,7 @@ namespace MoveToFileRefactorCommand {
 		readonly action: Proto.RefactorActionInfo;
 		readonly range: vscode.Range;
 		readonly trigger: vscode.CodeActionTriggerKind;
+		readonly targetFile?: string;
 	}
 }
 
@@ -142,7 +143,7 @@ class MoveToFileRefactorCommand implements Command {
 			return;
 		}
 
-		const targetFile = await this.getTargetFile(args.document, file, args.range);
+		const targetFile = args.targetFile;
 		if (!targetFile || targetFile.toString() === file.toString()) {
 			return;
 		}
