diff --git a/src/languageFeatures/updatePathsOnRename.ts b/src/languageFeatures/updatePathsOnRename.ts
index bdaa1fc..3c3e9a8 100644
--- a/src/languageFeatures/updatePathsOnRename.ts
+++ b/src/languageFeatures/updatePathsOnRename.ts
@@ -98,13 +98,13 @@ class UpdateImportsOnFileRenameHandler extends Disposable {
 			const resourcesBeingRenamed: vscode.Uri[] = [];
 
 			for (const { oldUri, newUri, newFilePath, oldFilePath, jsTsFileThatIsBeingMoved } of group) {
-				const document = await vscode.workspace.openTextDocument(jsTsFileThatIsBeingMoved);
+				// const document = await vscode.workspace.openTextDocument(jsTsFileThatIsBeingMoved);
 
 				// Make sure TS knows about file
 				this.client.bufferSyncSupport.closeResource(oldUri);
-				this.client.bufferSyncSupport.openTextDocument(document);
+				// this.client.bufferSyncSupport.openTextDocument(document);
 
-				if (await this.withEditsForFileRename(edits, document, oldFilePath, newFilePath)) {
+				if (await this.withEditsForFileRename(edits, oldFilePath, newFilePath)) {
 					resourcesBeingRenamed.push(newUri);
 				}
 			}
@@ -204,8 +204,7 @@ class UpdateImportsOnFileRenameHandler extends Disposable {
 		}
 
 		if (await isDirectory(resource)) {
-			const files = await vscode.workspace.findFiles(new vscode.RelativePattern(resource, '**/*.{ts,tsx,js,jsx}'), '**/node_modules/**', 1);
-			return files[0];
+			return resource;
 		}
 
 		return (await this._handles(resource)) ? resource : undefined;
@@ -213,12 +212,12 @@ class UpdateImportsOnFileRenameHandler extends Disposable {
 
 	private async withEditsForFileRename(
 		edits: vscode.WorkspaceEdit,
-		document: vscode.TextDocument,
+		// document: vscode.TextDocument,
 		oldFilePath: string,
 		newFilePath: string,
 	): Promise<boolean> {
 		const response = await this.client.interruptGetErr(() => {
-			this.fileConfigurationManager.setGlobalConfigurationFromDocument(document, nulToken);
+			// this.fileConfigurationManager.setGlobalConfigurationFromDocument(document, nulToken);
 			const args: Proto.GetEditsForFileRenameRequestArgs = {
 				oldFilePath,
 				newFilePath,
