diff --git a/src/tsServer/versionManager.ts b/src/tsServer/versionManager.ts
index 68d44a7..09880e9 100644
--- a/src/tsServer/versionManager.ts
+++ b/src/tsServer/versionManager.ts
@@ -76,19 +76,29 @@ export class TypeScriptVersionManager extends Disposable {
 		this._currentVersion = this.versionProvider.bundledVersion;
 	}
 
+	private pickItemToMessage(item: QuickPickItem): vscode.MessageItem & { run: any } {
+		let title = item.label;
+		if (item.description) {
+			title += ` (${item.description})`
+		}
+		if (item.detail) {
+			title += ` [${item.detail}]`
+		}
+		return {
+			title,
+			run: item.run,
+		}
+	}
+
 	public async promptUserForVersion(): Promise<void> {
-		const selected = await vscode.window.showQuickPick<QuickPickItem>([
-			this.getBundledPickItem(),
-			...this.getLocalPickItems(),
-			{
-				kind: vscode.QuickPickItemKind.Separator,
-				label: '',
-				run: () => { /* noop */ },
-			},
-			LearnMorePickItem,
-		], {
-			placeHolder: vscode.l10n.t("Select the TypeScript version used for JavaScript and TypeScript language features"),
-		});
+		const selected = await vscode.window.showInformationMessage(
+			vscode.l10n.t("Select the TypeScript version used for JavaScript and TypeScript language features"),
+			...[
+				this.getBundledPickItem(),
+				...this.getLocalPickItems(),
+				LearnMorePickItem,
+			].map(this.pickItemToMessage),
+		)
 
 		return selected?.run();
 	}
