diff --git a/src/tsServer/versionManager.ts b/src/tsServer/versionManager.ts
index c09e51e..f94ca31 100644
--- a/src/tsServer/versionManager.ts
+++ b/src/tsServer/versionManager.ts
@@ -77,6 +77,20 @@ export class TypeScriptVersionManager extends Disposable {
 		this._currentVersion = this.versionProvider.bundledVersion;
 	}
 
+	private pickItemToMessage(item: QuickPickItem): vscode.MessageItem & { run: any } {
+		let title = item.label;
+		if (item.description) {
+			title += ` (${item.description})`
+		}
+		if (item.detail) {
+			title += ` [${item.detail}]`
+		}
+		return {
+			title,
+			run: item.run,
+		}
+	}
+
 	public async promptUserForVersion(): Promise<void> {
 		const nativePreviewItem = this.getNativePreviewPickItem();
 		const items: QuickPickItem[] = [
@@ -89,17 +103,13 @@ export class TypeScriptVersionManager extends Disposable {
 		}
 
 		items.push(
-			{
-				kind: vscode.QuickPickItemKind.Separator,
-				label: '',
-				run: () => { /* noop */ },
-			},
 			LearnMorePickItem,
 		);
 
-		const selected = await vscode.window.showQuickPick<QuickPickItem>(items, {
-			placeHolder: vscode.l10n.t("Select the TypeScript version used for JavaScript and TypeScript language features"),
-		});
+		const selected = await vscode.window.showInformationMessage(
+			vscode.l10n.t("Select the TypeScript version used for JavaScript and TypeScript language features"),
+			...items.map(this.pickItemToMessage),
+		);
 
 		return selected?.run();
 	}
