diff --git a/src/languageFeatures/semanticTokens.ts b/src/languageFeatures/semanticTokens.ts
index 7f8d60d..0d9014a 100644
--- a/src/languageFeatures/semanticTokens.ts
+++ b/src/languageFeatures/semanticTokens.ts
@@ -20,7 +20,10 @@ export function register(
 		requireSomeCapability(client, ClientCapability.Semantic),
 	], () => {
 		const provider = new DocumentSemanticTokensProvider(client);
-		return vscode.languages.registerDocumentRangeSemanticTokensProvider(selector.semantic, provider, provider.getLegend());
+		return vscode.Disposable.from(
+			vscode.languages.registerDocumentRangeSemanticTokensProvider(selector.semantic, provider, provider.getLegend()),
+			vscode.languages.registerDocumentSemanticTokensProvider(selector.semantic, provider, provider.getLegend()),
+		);
 	});
 }
 
